action.php

```diff
@@ -50,6 +49,55 @@ public function handle_ajax_call_unknown(Doku_Event &$event, $param) {
         echo 'counted';
     }
 
-}
+    /**
+     * Fetch and add top pages to FeedCreator.
+     *
+     * Supported feed parameters:
+     * - mode: plugin-top (required)
+     * - lang: optional
+     * - month: optional (YYYYMM)
+     * - num: optional number of results (default = 10)
+     *
+     * @param Doku_Event $event
+     */
+    public function handleFeed(Doku_Event $event)
+    {
+        $opt = $event->data['opt'];
+        if ($opt['feed_mode'] !== 'plugin-top') return;
+
+        $event->preventDefault();
+
+        // set defaults as expected by the helper's best() method
+        $lang = isset($opt['lang']) ? $opt['lang'] : null;
+        $month = isset($opt['month']) ? $opt['month'] : null;
+        $num = isset($opt['num']) ? $opt['num'] : 10;
+
+        /** @var helper_plugin_top $hlp */
+        $hlp = plugin_load('helper', 'top');
+        $pages = $hlp->best($lang, $month, $num);
+
+        if (empty($pages)) return;
 
+        foreach ($pages as $page) {
+            $event->data['data'][] = [
+                'id' => $page['page'],
+                'score' => $page['value'],
+            ];
+        }
+    }
+
+    /**
+     * Set our data in feed item
+     *
+     * @param Doku_Event $event
+     */
+    public function handleFeedItemAdd(Doku_Event $event)
+    {
+        if ($event->data['opt']['feed_mode'] !== 'plugin-top') return;
+
+        $event->data['item']->title = $event->data['ditem']['id'];
+        $event->data['item']->link = hsc(wl($event->data['ditem']['id'], '', true));
+        $event->data['item']->description = $event->data['ditem']['score'];
```

Comment:
The title and description values are not optimal. The title should be the first header, the description should either be the fully rendered HTML of the page or the abstract as collected by `p_get_metadata($id,'description abstract')`;

Reply:
Are you sure? I actually decided to overwrite those feed defaults, because I did not want to potentially leak content protected by ACLs.

Reply:
feeds should check ACLs. If the helper not already does it, it needs to be added.